Content/Bosses/GlassMiniboss/Projectiles.GlassSpear.cs

```diff
@@ -38,16 +49,25 @@ public override void SetDefaults()
 
 		public override void OnSpawn(IEntitySource source)
 		{
-			Helpers.Helper.PlayPitched("GlassMiniboss/WeavingLong", 1f, 0f, Projectile.Center);
+			isMagmaVariant = setMagmaVariant;
+			setMagmaVariant = false;
 		}
 
 		public override void AI()
 		{
+			if (!isLoaded)
+			{
+				Helpers.Helper.PlayPitched("GlassMiniboss/WeavingLong", 1f, 0f, Projectile.Center);
+				isLoaded = true;
+			}
+
 			Timer++;
 
 			if (!Parent.active || Parent.type != NPCType<Glassweaver>())
 				Projectile.Kill();
 
+			Glassweaver glassweaver = Parent.ModNPC as Glassweaver;
```

Comment:
as above, should be null checked incase the boss despawns

Reply:
this does it happens on lines 66 and 67. it'll never be null but it could be inactive or not a glassweaver

Reply:
if its not a glassweaver, the ``as`` cast will result in null